lib/model/model.go

```diff
@@ -2099,18 +2099,34 @@ func unifySubs(dirs []string, exists func(dir string) bool) []string {
 
 	// Trim each item to itself or its closest known parent
 	for _, sub := range dirs {
-		for sub != "" && sub != ".stfolder" && sub != ".stignore" {
-			if exists(sub) {
-				break
+		if sub == "" || sub == "." || sub == string(filepath.Separator) {
+			// Shortcut. We are going to scan the full folder, so we can
+			// just return an empty list of subs at this point.
+			return nil
+		} else if exists(sub) {
+			l.Debugln("The sub " + sub + " exists/indexed and will be scanned")
+			subs = append(subs, sub)
+		} else {
+			l.Debugln("The sub " + sub + " is not indexed")
+			var subdir string
+			//initialising subdir and looping till an already indexed dir is found, The last unindexed dir/file is returned
```

Comment:
Please space after // like everywhere else, and sentences start with a capital letter.